exonum/src/api/mod.rs

```diff
@@ -40,38 +39,46 @@ use crate::{api::node::SharedNodeState, blockchain::Blockchain};
 
 mod with;
 
+/// Mutability of the endpoint. Used for auto-generated endpoints, e.g.
+/// in `moved_permanently` method.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum EndpointMutability {
+    /// Endpoint should process POST requests.
+    Mutable,
+    /// Endpoint should process GET requests.
+    Immutable,
+}
+
 /// This trait is used to implement an API backend for Exonum.
 pub trait ApiBackend: Sized {
     /// Concrete endpoint handler in the backend.
     type Handler;
     /// Concrete backend API builder.
     type Backend;
 
-    /// Add the given endpoint handler to the backend.
-    fn endpoint<N, Q, I, R, F, E>(&mut self, name: N, endpoint: E) -> &mut Self
+    /// Adds the given endpoint handler to the backend.
+    fn endpoint<Q, I, R, F, E>(&mut self, name: &'static str, endpoint: E) -> &mut Self
```

Comment:
Is `'static` lifetime necessary? Adding endpoints with dynamic naming is a valid use case, IMO.